idp/client.ts

```diff
@@ -0,0 +1,49 @@
+import { ScheduledEvent, Context } from "aws-lambda";
+import axios from "axios";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { SecretsManager } from "@aws-sdk/client-secrets-manager";
+const logger = new Logger({ serviceName: "TestIdp" });
+
+const secrets = new SecretsManager({ useFipsEndpoint: true });
+
+const IDP_CLIENT_ID = process.env.IDP_CLIENT_ID!;
+const IDP_CLIENT_SECRET_NAME = process.env.IDP_CLIENT_SECRET_NAME!;
+const IDP_DOMAIN = process.env.IDP_DOMAIN!;
+
+// We use the ScheduledEvent here but in actuality, we don't care about the type
+// of the event. This function is only ever going to invoked via the CLI or console
```

Comment:
Is the focus on `logger.info` or `logger.error` output directly due to expectation of invocation only from CLI or console?

Reply:
Yeah, pretty much. We could probably type `event` as `any` or `{}` but originally I had thought I'd setup a trigger to just call it every so often for fun.

But yeah the data never gets _sent_ anywhere since there's not really any way to get the return value (it's not returned to another service like API Gateway or Step Functions). The real expectation is that this is being invoked just to "prove out" the process and maybe only once ever in it's life.

It is possible to get the payload from a `lambda:Invoke` via the CLI iirc but honestly, that feels hardly worth it.

So yeah, I'm making people go dig in the logs mostly because it's easier to implement and more consistent with what we'll need to do someday anyway. Because soon, this function won't exist. We'll turn it into a library that other functions can leverage as needed (probably very soon actually, in AT-7337). And no Function should ever _return_ any information about its token (and we assume in our accounts if you can directly Invoke a Lambda you probably can also read its logs).